### PR TITLE
feat(path): flexible matching for mapped-locations

### PR DESCRIPTION
**Pull request by asherber**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

I'm not sure how you'll feel about this one. The goal was to make `mapped_locations` more portable. 

My use case is that I've got two machines with similar directory structures, except that I'm `Aaron` on one and `ASherber` on the other. I'd like to be able to put `"~\\Documents\\Projects"` in `mapped_locations` and have it work on both machines (i.e., match both `C:\Users\Aaron\Documents\Projects` and `C:\Users\ASherber\Documents\Projects`). 

While I'm at it, because I hate having to escape backslashes, and because Windows doesn't care which path separator I use, I'd like to be able to use `"~/Documents/Projects"`. And then if I make `mapped_locations` case-insensitive on Windows and Mac, I can actually use `"~/documents/projects"` and use the same theme file on all OSes.

If you agree with this approach, I've got similar changes queued up for `include_folders` and `exclude_folders` which I could either add to this PR or put into a new one.

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: docs/docs/segment-path.md [hunk 1]
```diff
@@ -43,12 +43,12 @@ Display the current path.
 ## Mapped Locations
 
 Allows you to override a location with an icon. It validates if the current path **starts with** the value and replaces
-it with the icon if there's a match. To avoid issues with nested overrides, Oh my posh will sort the list of mapped
+it with the icon if there's a match. To avoid issues with nested overrides, oh-my-posh will sort the list of mapped
```
Comment (JanDeDobbeleer) [inline docs/docs/segment-path.md]: The docs should always refer to Oh My Posh, to stay consistent.

File: docs/docs/segment-path.md [hunk 2]
```diff
@@ -43,12 +43,12 @@ Display the current path.
...
 locations before doing a replacement.
 
 - mapped_locations_enabled: `boolean` - replace known locations in the path with the replacements before applying the
-style. defaults to `true`
-- mapped_locations: `map[string]string` - custom glyph/text for specific paths. Works regardless of the `mapped_locations_enabled`
+style - defaults to `true`
+- mapped_locations: `object` - custom glyph/text for specific paths. Works regardless of the `mapped_locations_enabled`
```
Comment (JanDeDobbeleer) [inline docs/docs/segment-path.md:51]: Why did you change this to object?

File: src/segment_path.go [hunk 3]
```diff
@@ -220,6 +220,19 @@ func (pt *path) getPwd() string {
 	return pwd
 }
 
+func (pt *path) normalize(inputPath string) string {
```
Comment (JanDeDobbeleer) [inline src/segment_path.go:223]: this would need to be put outside of the scope of path when you want to reuse the functionality.

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by JanDeDobbeleer** [comment]
Agreed with doing this on macOS and Windows given their nature. Actually like this, amazing flexibility and a real world use-case. Also well written case on why and how. Great work overall.

File: docs/docs/segment-path.md (hunk 1 above)
Comment (asherber) [reply docs/docs/segment-path.md]: Okay -- I thought I'd seen it a few ways in the current docs, and the current "Oh my posh" didn't seem right. I'll change this.

File: docs/docs/segment-path.md (hunk 2 above)
Comment (asherber) [reply docs/docs/segment-path.md:51]: Because it's an object in JSON, which is what we care about here, and a map in Go. This specification confused me when I first saw it. `mapped_shell_names` has the same structure, and it's specified as `object`.

File: src/segment_path.go (hunk 3 above)
Comment (asherber) [reply src/segment_path.go:223]: Yes, and possibly also refactored a bit.

File: docs/docs/segment-path.md (hunk 1 above)
Comment (JanDeDobbeleer) [reply docs/docs/segment-path.md]: If we now align I do believe oh-my-posh makes more sense, but that's beyond the scope of this PR IMHO.

File: src/segment_path.go (hunk 3 above)
Comment (JanDeDobbeleer) [reply src/segment_path.go:223]: I suppose we leave that refactoring for the following PR?
